No point harrassing the users in case of broken protocol

diff -BurpN rp-pppoe.orig/src/plugin.c rp-pppoe/src/plugin.c
--- rp-pppoe.orig/src/plugin.c	2009-05-05 19:42:36.000000000 +0000
+++ rp-pppoe/src/plugin.c	2009-05-05 19:44:56.000000000 +0000
@@ -65,6 +65,9 @@ char pppd_version[] = VERSION;
 /* From sys-linux.c in pppd -- MUST FIX THIS! */
 extern int new_style_driver;
 
+/* Supplied by pppd */
+extern int debug;
+
 char *pppd_pppoe_service = NULL;
 static char *acName = NULL;
 static char *existingSession = NULL;
@@ -217,7 +220,7 @@ PPPOESendConfig(int mtu,
     struct ifreq ifr;
 
     if (mtu > MAX_PPPOE_MTU) {
-	warn("Couldn't increase MTU to %d", mtu);
+	if (debug) warn("Couldn't increase MTU to %d", mtu);
 	mtu = MAX_PPPOE_MTU;
     }
     sock = socket(AF_INET, SOCK_DGRAM, 0);
@@ -241,7 +244,7 @@ PPPOERecvConfig(int mru,
 		int pcomp,
 		int accomp)
 {
-    if (mru > MAX_PPPOE_MTU) {
+    if (mru > MAX_PPPOE_MTU && debug) {
 	warn("Couldn't increase MRU to %d", mru);
     }
 }
